Modules/CaPTkInteractiveSegmentation/include/CaPTkInteractiveSegmentation.h

```diff
@@ -40,16 +33,26 @@ class MITKCAPTKINTERACTIVESEGMENTATION_EXPORT CaPTkInteractiveSegmentation /*fin
      * algorithm finishes, OnAlgorithmFinished() is called.
      * 
      * @param images a list of the co-registered input images
-     * @param labels label image that contains the user drawn seeds
+     * @param seeds labels image that contains the user drawn seeds
     */
     void Run(std::vector<mitk::Image::Pointer>& images, 
              mitk::LabelSetImage::Pointer& seeds);
 
-    // void Run(Json::Value& task_json, Json::Value& cohort_json);
+signals:
+    /** \brief This gets emitted when the async execution finishes */
+    void finished(bool ok, std::string errorMessage, mitk::LabelSetImage::Pointer result);
+
+    void progressUpdate(int progress);
 
-    // void Run(std::string task_json_path, std::string cohort_json_path);
+protected slots:
+    /** \brief This function runs in the main thread when 
+     * the algorithm is finished to add the result to the data storage
+    */
+    void OnAlgorithmFinished();
 
-    void SetProgressBar(QProgressBar* progressBar);
+    void OnProgressUpdateInternalReceived(int progress);
```

Comment:
What is the difference between the 2 update methods?

Reply:
Instead of connecting the ProgressUpdate(int) signal, with the QProgressBar's setValue(int) slot, we connect it with OnProgressUpdateInternalReceived(int), which then emits the signal. This re-emitted signal gets connected with the progress bar in the plugin.

That intermediate stage happens because the initial ProgressUpdate(int) signal doesn't belong to the class captk::InteractiveSegmentation, but a class it uses, so we can't connect it directly.